hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```
@@ -267,21 +267,27 @@ public void deleteCompactionRequested(HoodieInstant instant) {
   }
 
   public void deleteInstantFileIfExists(HoodieInstant instant) {
+    deleteInstantFileIfExists(instant, true);
+  }
+
+  public boolean deleteInstantFileIfExists(HoodieInstant instant, boolean exceptionIfFailToDelete) {
     LOG.info("Deleting instant " + instant);
-    Path inFlightCommitFilePath = getInstantFileNamePath(instant.getFileName());
+    Path commitFilePath = getInstantFileNamePath(instant.getFileName());
     try {
-      if (metaClient.getFs().exists(inFlightCommitFilePath)) {
-        boolean result = metaClient.getFs().delete(inFlightCommitFilePath, false);
+      if (metaClient.getFs().exists(commitFilePath)) {
+        boolean result = metaClient.getFs().delete(commitFilePath, false);
         if (result) {
           LOG.info("Removed instant " + instant);
-        } else {
+        } else if (exceptionIfFailToDelete) {
```

again, i don't think we should allow silencing exception at this level